### PR TITLE
16 engagement

### PR DESCRIPTION
**Pull request by nghe0001**
Worked on the form with Shawn

# Description
- Engagment Form created
- Contact list mockup (-Todo not calling database) 
- Select Contact component added with table( 2 for now)
- Axios post with form to Engagement Collection (no validation yet)
- End point for engagement created

Todo - 
     refine css,
     Add tag collection for separate axios post, 
     Will update code to use more component,
     Api call for contact to be populated on the contact list, 

## Test Instructions

Go to localhost:3000/en/add/engagement to test form
Fill in the form

## Help Requested
Got Help from Shawn, Jordan, Marco, Ben , Adam

## Checklist
- [ ] Unit tests have been added/updated

### CODE REVIEW

**Comment by will0684** [comment]
@nghe0001 seeder was missing a couple fields for the engagements (my bad), mind if I add a commit to this PR? @nghe0001

File: components/engagement/EngForm.vue [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+              </div>
+            </div>
+          </div>
+
+          <div class="max-w-lg sm:w-1/3 mb-4">
+            <label
+              class="orange block tracking-wide text-black text-md font-bold mb-2"
+              for="numParticipants"
+            >
+              Number of numParticipants
```
Comment (andr0272) [inline components/engagement/EngForm.vue:93]: Typo?

File: components/engagement/EngForm.vue [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+            </div>
+          </div>
+        </div>
+        <div class="flex flex-wrap">
+          <div class="w-full sm:w-6/12 mb-8">
+            <label
+              class="orange block tracking-wide text-black text-md font-bold -mb-4"
+              for="description"
+            >
+              Description maximum 1000 characters. Additional characters will not be
```
Comment (andr0272) [inline components/engagement/EngForm.vue:119]: Typo?

File: components/engagement/EngForm.vue [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      if (this.engagementDetail.numParticipants > 0) {
+        this.engagementDetail.numParticipants--
+      }
+    },
+    // dateAdj() {
+    //   return this.engagementDetail.date.setDate(
+    //     this.engagementDetail.date.getDate() + 1
+    //   )
+    // },
+    // eslint-disable-next-line space-before-function-paren
```
Comment (andr0272) [inline components/engagement/EngForm.vue:289]: This has been fixed. No need to ignore this any more

File: components/engagement/EngForm.vue [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+    notification(type, message) {
+      this.message.type = type
+      this.message.message = message
+      this.timeout = setTimeout(() => this.clearMessage(), 5000)
+    },
+
+    // eslint-disable-next-line space-before-function-paren
```
Comment (andr0272) [inline components/engagement/EngForm.vue:296]: This has been fixed. No need to ignore this any more

File: components/engagement/EngForm.vue [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+    },
+
+    async submitForm(engagementDetail) {
+      try {
+        await this.$axios.post('/api/engagement/addEngagement', {
+          engagementDetail
+        })
+        this.notification('success', 'engagment created')
+      } catch (e) {
+        // eslint-disable-next-line no-console
```
Comment (andr0272) [inline components/engagement/EngForm.vue:310]: :eyes:

File: components/engagement/EngShowContacts.vue [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+</template>
+
+<script>
+export default {
+  props: {
+    contactName: { type: String, default: '' },
+    department: { type: String, default: '' },
+    contactEmail: { type: String, default: '' },
+    lastEngTitle: { type: String, default: '' },
+    engagementDate: { type: String, default: '' }, // data type should be Date
```
Comment (andr0272) [inline components/engagement/EngShowContacts.vue:25]: ?

**Pull request merged**